projects/hipsparse/CMakeLists.txt

```
@@ -1,289 +1,278 @@
-option(BUILD_CODE_COVERAGE "Build with code coverage enabled" OFF)
-option(BUILD_ADDRESS_SANITIZER "Build with address sanitizer enabled" OFF)
-option(BUILD_DOCS "Build documentation" OFF)
-
-if(BUILD_CUDA)
-  message(DEPRECATION "Using BUILD_CUDA is deprecated and will be removed as an option in a future release. Instead use USE_CUDA")
-  set(USE_CUDA ${BUILD_CUDA})
+if(HIPSPARSE_ENABLE_ASAN AND HIPSPARSE_ENABLE_FORTRAN)
+    message(FATAL_ERROR "Cannot enable both address sanitizer and Fortran clients")
 endif()
 
-if(BUILD_CODE_COVERAGE)
-  add_compile_options(-fprofile-arcs -ftest-coverage)
-  add_link_options(--coverage)
+# Set default build type
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Release' as none was specified.")
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "" "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-if(BUILD_ADDRESS_SANITIZER)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address -shared-libasan")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address -shared-libasan")
-  add_link_options(-fuse-ld=lld)
+# Conditionally enable Fortran
+# NOTE: Behavior change - Fortran now controlled by HIPSPARSE_ENABLE_FORTRAN option (default ON)
+# Previously enabled automatically when building tests or benchmarks
+if(HIPSPARSE_ENABLE_FORTRAN)
+    enable_language(Fortran)
 endif()
 
-# Dependencies
-include(cmake/Dependencies.cmake)
+# --- Dependency Setup ---
 
-# Setup version
-rocm_setup_version(VERSION 4.2.0)
-set(hipsparse_SOVERSION 4.2.0)
+find_package(Python3 3.9 COMPONENTS Interpreter REQUIRED)
 
-# hipSPARSE library
-if (NOT BUILD_CLIENTS_ONLY)
-  add_subdirectory(library)
+if(HIPSPARSE_ENABLE_CUDA)
+    find_package(HIP REQUIRED CONFIG)
+    # list(APPEND HIP_INCLUDE_DIRS "${HIP_ROOT_DIR}/include")
+    find_package(CUDA REQUIRED)
+elseif(HIPSPARSE_ENABLE_HIP)
+    find_package(hip REQUIRED CONFIG)
+    find_package(rocsparse 4.0.1 REQUIRED CONFIG)
 endif()
 
-if(BUILD_CLIENTS_ONLY OR BUILD_CLIENTS_SAMPLES OR BUILD_CLIENTS_TESTS OR BUILD_CLIENTS_BENCHMARKS)
-  if(NOT CLIENTS_OS)
-    rocm_set_os_id(CLIENTS_OS)
-    string(TOLOWER "${CLIENTS_OS}" CLIENTS_OS)
-    rocm_read_os_release(CLIENTS_OS_VERSION VERSION_ID)
-  endif()
-  message(STATUS "OS: ${CLIENTS_OS} ${CLIENTS_OS_VERSION}")
-  set(GFORTRAN_RPM "libgfortran4")
-  set(GFORTRAN_DEB "libgfortran4")
-  find_package(OpenMP QUIET)
-  if(OPENMP_FOUND)
-    set(OPENMP_RPM "libgomp")
-    set(OPENMP_DEB "libomp-dev")
-  endif()
-  if(CLIENTS_OS STREQUAL "centos" OR CLIENTS_OS STREQUAL "rhel" OR CLIENTS_OS STREQUAL "almalinux")
-    if(CLIENTS_OS_VERSION VERSION_GREATER_EQUAL "8")
-      set(GFORTRAN_RPM "libgfortran")
+if(HIPSPARSE_ENABLE_CLIENT OR HIPSPARSE_BUILD_TESTING OR BUILD_TESTING)
+    if(HIPSPARSE_ENABLE_OPENMP)
+        find_package(OpenMP REQUIRED)
     endif()
```

The following will prefer to use rocm-omp if it can find it and then fall back to gomp if not.

```suggestion
if(HIPSPARSE_ENABLE_OPENMP)
  find_package(OpenMP CONFIG HINTS llvm/lib/cmake)
  add_library(hipsparse-omp-adapter INTERFACE)
  add_library(hipsparse::omp-adapter ALIAS tensile-omp-adapter)
  if(OpenMP_FOUND)
    set_target_properties(hipsparse-omp-adapter
      PROPERTIES
        INTERFACE_COMPILE_OPTIONS -fopenmp
        INTERFACE_LINK_LIBRARIES OpenMP::omp
    )
  else()
    message(STATUS "Did not find rocm-omp - falling back to gomp")
    find_package(OpenMP REQUIRED)
    target_link_libraries(hipsparse-omp-adapter INTERFACE OpenMP::OpenMP_CXX)
  endif()
```